tensorflow/java/src/gen/cc/op_gen_main.cc

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ==============================================================================*/
+
+#include <string>
+#include <vector>
+
+#include "tensorflow/core/platform/init_main.h"
+#include "tensorflow/core/platform/env.h"
+#include "tensorflow/core/util/command_line_flags.h"
+#include "tensorflow/core/lib/core/status.h"
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/java/src/gen/cc/op_generator.h"
+
+namespace tensorflow {
+namespace op_gen {
+
+const char kUsageHeader[] =
+    "\n\nGenerator of operation wrappers in Java.\n\n"
+    "This executable generates wrappers for all operations registered in the\n"
+    "ops file it has been linked to (i.e. one of the /core/ops/*.o binaries).\n"
+    "Generated files are output to the path provided as an argument, under\n"
+    "their appropriate package and using a maven-style directory layout.\n\n";
+
+}  // namespace op_gen
+}  // namespace tensorflow
+
+int main(int argc, char* argv[]) {
+  tensorflow::string ops_file;
+  tensorflow::string output_dir;
+  std::vector<tensorflow::Flag> flag_list = {
+    tensorflow::Flag("file", &ops_file,
```

Comment:
This sounds like a good idea. But it might lead to some other difficulties in other way. Let me explain.

The `ops_file` (or let rename it `ops_lib` or `ops_lib_name`) is not only used to produce a package name but also to extract the group name (`@Operator(group="...")`) in a near future, so it could be accessible from the `Ops` API (`ops.groupName().opName()`).

But the package and group names uses different naming conventions, the former being flat and the later in pascal/camel case. So by transforming, for instance, the `control_flow_ops` library name to `org.tensorflow.op.controlflow` in the Bazel rule, we won't be able to extract the `controlFlow` group name from it.

A solution could be to pass the `org.tensorflow.op` root package as an argument but also the group name as a different argument (`--group`), which could then be flatten to a fully qualified package name, what do you think? 

If we do so, then in addition to remove the `_op[s]` suffix to the lib name, I need to find out an easy way to transform it to camel case from its original underscore form.